### ISSUE TITLE
Is there a way to edit stop loss and take profit for open positions in bybit futures?

### ISSUE THREAD

**Issue by 19bk** [ccxt/ccxt#18257]
### ؜

OS: Ubuntu
Programming Language version: python: Python 3.10.6
CCXT version: CCXT Version: 3.1.40

I tried **edit_order** but it's not working. 

```
try:
    params = {
        # "stopLoss":myStopLoss, "takeProfit":myTakeProfit,
        'positionIdx': 1, # 0 One-Way Mode, 1 Buy-side, 2 Sell-side
    }
    order = exchange.create_order(symbol=symbol, type='Market', side='buy', amount=amount, params=params)
    order_id = order['id']
    print(order)
    # print(order_id)
    open_orders = exchange.fetchPosition (symbol, params = {}) 
    print(open_orders['entryPrice'])
    
    exchange.edit_order(id=order_id,
                    side="buy",
                    symbol=symbol,
                    type="market",
                    params={
                        'stop_loss': (open_orders['entryPrice'] * 0.99)
                    })
except Exception as e:
    print(e)
```

**Comment by sc0Vu** [comment]
Hi @19bk 

From bybit documentation, you can only modify unfilled or partially filled orders. (https://bybit-exchange.github.io/docs/v5/order/amend-order)

To update tp/sl for the open position, I found this api: https://bybit-exchange.github.io/docs/v5/position/trading-stop

**Comment by kaanguven** [comment]
> Hi @19bk 
> 
> From bybit documentation, you can only modify unfilled or partially filled orders. (https://bybit-exchange.github.io/docs/v5/order/amend-order)
> 
> To update tp/sl for the open position, I found this api: https://bybit-exchange.github.io/docs/v5/position/trading-stop

Any update?

**Comment by 19bk** [comment]
Thanks for the tip  @sc0Vu! the API works, https://bybit-exchange.github.io/docs/v5/position/trading-stop

**Comment by svax974** [comment]
Hello, @19bk how do you use that API endpoint through CCXT ?

**Comment by 19bk** [comment]
For every symbol you can only have one open position in a single category e.g linear. However you can multiple orders for the same symbol.

**Comment by svax974** [comment]
Sorry this is a bit off-topic.
@19bk Ok, is this the same on all exchanges ? Do they all consider only one position per market (and category) ?

**Comment by 19bk** [comment]
Here is some more context to the position modes.
One-Way Mode: In this mode, you can only open one position per symbol, either on the Buy or Sell side. This is a more traditional approach where if you open a new position in the same market, it modifies your existing position.

Hedge Mode: If you are in hedge mode, you can open both Buy and Sell side positions simultaneously for the same symbol. This allows you to have multiple open positions in the same market, effectively enabling you to maintain long and short positions at the same time

**Comment by CryptoDjam** [comment]
modif_order = exchange.fetch_open_orders(symbol)

            for order in modif_order:
                if order['side'] == 'sell':
                    order_id = order['id']
                    print(order_id)

 
                    exchange.edit_order(order_id, symbol, 'market', 'buy', 0.1, None, {'stopLossPrice': new_stop_price})

**Comment by CryptoDjam** [comment]
modif_order = exchange.fetch_open_orders(symbol)

            for order in modif_order:
                if order['side'] == 'sell':
                    order_id = order['id']
                    print(order_id)

      
   
                    exchange.edit_order(order_id, symbol, 'market', 'buy', 0.1, None, {'stopLossPrice': new_stop_price})

**Comment by ttodua** [comment]
late reply, but since then we had started unification for stoploss/takeprofit orders.
for example, see this documentation to understand the details: https://docs.ccxt.com/#/README?id=stop-loss-orders

so, as long as user wants to add separate SL order for existing position, you can submit a separate order as described there.  if `stopLossPrice` is not supported by exchange features (check by [this way](https://docs.ccxt.com/#/README?id=features)) then use regular "[trigger order](https://docs.ccxt.com/#/README?id=conditional-orders)" with 'reduceOnly = true' param.


let us know if anyone has any issues

**Comment by kaanguven** [comment]
> late reply, but since then we had started unification for stoploss/takeprofit orders. for example, see this documentation to understand the details: https://docs.ccxt.com/#/README?id=stop-loss-orders
> 
> so, as long as user wants to add separate SL order for existing position, you can submit a separate order as described there. if `stopLossPrice` is not supported by exchange features (check by [this way](https://docs.ccxt.com/#/README?id=features)) then use regular "[trigger order](https://docs.ccxt.com/#/README?id=conditional-orders)" with 'reduceOnly = true' param.
> 
> let us know if anyone has any issues

So, can we now send a combined order (containing TP and SL values) like the one below (for example, for Binance)?

params = {
'stopLossPrice': 55.45, 
'takeProfitPrice': 120.45,
}

order = exchange.create_order (symbol, type, side, amount, price, params)

**Comment by ttodua** [comment]
> > late reply, but since then we had started unification for stoploss/takeprofit orders. for example, see this documentation to understand the details: [docs.ccxt.com#/README?id=stop-loss-orders](https://docs.ccxt.com/#/README?id=stop-loss-orders)
> > so, as long as user wants to add separate SL order for existing position, you can submit a separate order as described there. if `stopLossPrice` is not supported by exchange features (check by [this way](https://docs.ccxt.com/#/README?id=features)) then use regular "[trigger order](https://docs.ccxt.com/#/README?id=conditional-orders)" with 'reduceOnly = true' param.
> > let us know if anyone has any issues
> 
> So, can we now send a combined order (containing TP and SL values) like the one below (for example, for Binance)?
> 
> params = { 'stopLossPrice': 55.45, 'takeProfitPrice': 120.45, }
> 
> order = exchange.create_order (symbol, type, side, amount, price, params)

no, for attaching it to opening order, you need this https://docs.ccxt.com/#/README?id=stoploss-and-takeprofit-orders-attached-to-a-position
